enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,613 @@
+workflows will be present in the mirror.  More details about the
+various services follow below.
+
+The image generated by the `agent create-image` command will also
+contain the data in the input `install-config.yaml`,
+`agent-config.yaml`, and other files. That data may be converted to
+another format (we anticipate using the ZTP CRDs, since those map to
+the assisted-service API calls). The intermediate format will be
+considered another layer of control for the cluster creator, just as
+the manifests written by the `create manifests` sub-command are. The
+cluster creator will be able to generate those intermediate files with
+the sub-command `agent create-manifests`.
+
+One host, designated as "node0", will be configured to run the
+assisted service and a client that uses the assisted service REST API
+to automate the deployment. Because we must support single-node and
+compact deployments, the node selected as node0 must always be
+configured as a control plane node. Selecting other control plane
+nodes may be left to the user.
+
+All hosts will run the assisted agent, configured to talk to the
+service on node0.
+
+When all of the hosts are booted with the image, the client on node0
+will use the assisted service to trigger the deployment of the cluster
+(including applicable pre-flight validations) without further
+interaction from the user.
+
+The `agent create-image` command will produce as output credentials to
+access the cluster that is about to be created. This allows a client,
+such as a tool monitoring deployment progress, to use the cluster's
+API as soon as the API service is available.
+
+During deployment, the third-party orchestration tool will need a way
+to monitor progress and discover errors. The assisted service
+read-only API endpoints running on node0 will report that data. In the
+event of failure, a read-only endpoint in that service will be
+available to download any logs that have been collected.
+
+We will be taking advantage of the assisted service's ability to run
+the bootstrap services "in place" to avoid having a separate bootstrap
+VM. Bootstrap will run on node0, where the assisted service runs, to
+ensure that the service and its API are available for as long as
+possible.
+
+When bootstrapping completes and node0 reboots, progress monitoring
+will need to shift to the OpenShift APIs (`ClusterVersion` and
+`ClusterOperator`). We will extend the `openshift-install` binary with
+a new sub-command, `agent wait-for`, to use both APIs for monitoring
+and [progress
+reporting](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/installation-progress-bar.md)
+in that case, including reporting when steps like validation fail and
+the cluster deployment is never going to succeed. *Users will still be
+able to write their own monitoring logic using the assisted service
+and OpenShift APIs directly, but they will have the option of using
```

Comment:
I think this will be so complicated that very few will do that (REST API, OpenShift API, switching from one to the other)

Reply:
I suspect you're right, that's why we're providing an implementation in the installer.